Source/ParameterEncoding.swift

```diff
@@ -161,16 +193,16 @@ public struct URLEncoding: ParameterEncoding {
             }
         } else if let array = value as? [Any] {
             for value in array {
-                components += queryComponents(fromKey: "\(key)[]", value: value)
+                components += queryComponents(fromKey: encodeArrayKey(key), value: value)
```

Comment:
Moving the `encode` functionality to `ArrayEncoding` would make this `arrayEncoding.encode(key)`, which I think is much clearer.